toolkit/docs/how_it_works/3_package_building.md

```diff
@@ -144,34 +148,20 @@ In the example below nodes `A-a` and `A-b` are from the same spec file and requi
-
-#### Trimming
-Depending on which packages have been selected for building only some parts of the full graph are needed. The `graphoptimizer` tool adds a `TypeGoal` node to the graph which depends on all the high level packages it has been asked to make available. It then creates a sub-graph rooted at that goal node and continues processing the sub-graph.
-
-#### Marking as Up-To-Date
-A node which is `TypeBuild`/`StateBuild` will be marked as `StateUpToDate` if `graphoptimizer` is able to find all the expected output packages with the correct version numbers. `Graphoptimizer` WILL NOT trigger rebuilds due to changes in the timestamps of SPEC files and packages. If a package must be rebuilt its base package name (name of SPEC file) should be passed via the variable `PACKAGE_REBUILD_LIST=`. This will cause `graphoptimizer` to never mark the packages as up-to-date.
-
-`Graphoptimizer` is also able to detect changes in dependencies which should trigger a rebuild of other packages. For example if package `A` depends on package `B`, and package `B` is rebuilt, then package `A` will also be marked for rebuild.
+#### Default Goal Node
+The `grapher` tool automatically adds an "ALL" goal node to the graph which links to every node. Building this node will cause every known package to be built.
 
-Sometimes packages fail to build correctly, or are otherwise not suitable for building in the build system. `Graphoptimizer` would normally complain that they are missing, possibly causing undesired rebuilds. The `PACKAGE_IGNORE_LIST=` variable can be used to set the `--ignore-packages` argument which instructs `graphoptimizer` to never worry about the listed packages (again using the package base name).
 
-### Stage 3: Graphpkgfetcher
+### Stage 2: Graphpkgfetcher
 The `graphpkgfetcher` tool's job is to resolve unresolved remote nodes. Unresolved nodes occur when a local package has `Requires` or `BuildRequires` which are not available from another local package.
 
-The tool uses the `worker_chroot` (see [Chroot Worker](1_initial_prep.md#chroot_worker)) to locate packages. The worker will search in five locations: 1) the local chroot environment, 2) already build RPMs in `./../out/RPMS/`, 3) the upstream base repository 4) the upstream update repository if `$(USE_UPDATE_REPO)` is set to `y` 5) any remote repo listed in `REPO_LIST ?=`. If `$(DISABLE_UPSTREAM_REPOS)` is set to `y`, any repo that is accessed through the network is disabled.
+The tool uses the `worker_chroot` (see [Chroot Worker](1_initial_prep.md#chroot_worker)) to locate packages. The worker will search in three locations: 1) the local chroot environment, 2) already build RPMs in `./../out/RPMS/`, 3) any remote repo listed in `REPO_LIST ?= $(MANIFESTS_DIR)/package/base.repo`.
```

Comment:
Revert change here, new documentation was loss in the merge

Reply:
Note to self: review when the merge is fixed.

---
In reply to: [496089036](https://github.com/microsoft/CBL-Mariner/pull/141#discussion_r496089036) [](ancestors = 496089036)